From 0c874628796cb29558bafd605af45daf560e9400 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Tue, 14 Jan 2020 08:13:31 +0200
Subject: [PATCH] [SW-6739] arcboot: decrease heap and stack size

because of high utilization of DCCM by stack and heap
values where decreased to freeup DCCM for future use.
it was teseted by increasing the heap and stack till the
program execution completed successfully

Change-Id: I9587b7dccae112ce7a55657e772fd74b97747f30
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/goya2_arc/arcboot_rom.lds b/goya2_arc/arcboot_rom.lds
index 3500e00..2f46595 100644
--- a/goya2_arc/arcboot_rom.lds
+++ b/goya2_arc/arcboot_rom.lds
@@ -86,7 +86,7 @@
 	.heap : {
 		. = ALIGN(32 / 8);
 		__start_heap = . ;
-		. = . + (DEFINED(__HEAP_SIZE) ? __HEAP_SIZE : (DEFINED(__DEFAULT_HEAP_SIZE) ? __DEFAULT_HEAP_SIZE : 10k)) ;
+		. = . + (DEFINED(__HEAP_SIZE) ? __HEAP_SIZE : (DEFINED(__DEFAULT_HEAP_SIZE) ? __DEFAULT_HEAP_SIZE : 5k)) ;
 		__end_heap = . ;
 		. = ALIGN(0x8);
 	} > REGION_HEAP
@@ -94,7 +94,7 @@
 	.stack : {
 		. = ALIGN(32 / 8);
 		__stack = . ;
-		. = . + (DEFINED(__STACK_SIZE) ? __STACK_SIZE : (DEFINED(__DEFAULT_STACK_SIZE) ? __DEFAULT_STACK_SIZE : 3k)) ;
+		. = . + (DEFINED(__STACK_SIZE) ? __STACK_SIZE : (DEFINED(__DEFAULT_STACK_SIZE) ? __DEFAULT_STACK_SIZE : 1k)) ;
 		__stack_top = . ;
 	} > REGION_STACK
 }
\ No newline at end of file
